axelor-sale/src/main/java/com/axelor/apps/sale/web/ConfiguratorController.java

```diff
@@ -95,6 +100,36 @@ public void generateProduct(ActionRequest request, ActionResponse response) {
     }
   }
 
+  public void regenerateProduct(ActionRequest request, ActionResponse response) {
+    Configurator configurator = request.getContext().asType(Configurator.class);
+    JsonContext jsonAttributes = (JsonContext) request.getContext().get("$attributes");
+    JsonContext jsonIndicators = (JsonContext) request.getContext().get("$indicators");
+    configurator = Beans.get(ConfiguratorRepository.class).find(configurator.getId());
+    try {
+      Beans.get(ConfiguratorService.class)
+          .regenerateProduct(
+              configurator,
+              Beans.get(ProductRepository.class).find(configurator.getProduct().getId()),
+              jsonAttributes,
+              jsonIndicators,
+              getSaleOrderId(request.getContext()));
+      response.setReload(true);
+      if (configurator.getProduct() != null) {
+        response.setView(
+            ActionView.define(I18n.get("Product generated"))
+                .model(Product.class.getName())
+                .add("form", "product-form")
+                .add("grid", "product-grid")
+                .param("search-filters", "products-filters")
+                .context("_showRecord", configurator.getProduct().getId())
+                .map());
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(e);
+      response.setError(e.getMessage());
```

Comment:
Can be replaced by `TraceBackService.trace(response, e, ResponseMessageType.ERROR);`